### PR TITLE
Added v2 memory test

### PR DESCRIPTION
**Pull request by Zyqsempai**
@AkihiroSuda @crosbymichael PTAL, what do you think, can we move in that direction with unit tests? or you saw it in a different way?
Signed-off-by: bpopovschi <zyqsempai@mail.ru>

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
I think we should use real cgroup for testing

**Comment by Zyqsempai** [comment]
@AkihiroSuda You mean something like this? Check it out please, I updated code a little bit.

**Comment by Zyqsempai** [comment]
@AkihiroSuda I tried few approaches with `real cgroup` and every time struggling with permissions. I didn't find in any realization of cgroups real cgroup usage for tests.
How do you plan to do so? Coz I have no more ideas(I tried to mount it, tried to use existing one in /sys/fs/cgroup/unified)

File: v2/testutils_test.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+const defaultCgroup2Path = "/sys/fs/cgroup/unified"
```
Comment (AkihiroSuda) [inline v2/testutils_test.go]: Use `/sys/fs/cgroup`. You can just skip  the test when running in hybrid mode.

File: v2/memoryv2_test.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+			Max:  pointerInt64(629145600),
+			Swap: pointerInt64(314572800),
+			High: pointerInt64(524288000),
+		},
+	}
+	c, err := NewManager(cg.cgv2Root, "/memory-test-cg", &res)
+	if err != nil {
+		t.Fatal("failed to init new cgroup manager: ", err)
+	}
+	cg.writeFileContents(map[string]string{
```
Comment (AkihiroSuda) [inline v2/memoryv2_test.go]: toggle the controller and write the limit files via the manager, and then verify the files via `ioutil.ReadFile`, and then also verify the stat struct from the manager.

File: v2/memoryv2_test.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+		Pids: &Pids{},
+		IO:   &IO{},
+		RDMA: &RDMA{},
+		Memory: &Memory{
+			Max:  pointerInt64(629145600),
+			Swap: pointerInt64(314572800),
+			High: pointerInt64(524288000),
+		},
+	}
+	c, err := NewManager(cg.cgv2Root, cg.groupPath, &res)
```
Comment (AkihiroSuda) [inline v2/memoryv2_test.go]: `NewManager` should take `/sys/fs/cgroup` as the root

**Comment by Zyqsempai** [comment]
@AkihiroSuda @crosbymichael PTAL

File: v2/memoryv2_test.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCgroupv2MemoryStats(t *testing.T) {
+	cg := NewCgroupDir("/memory-test-cg", t)
```
Comment (AkihiroSuda) [inline v2/memoryv2_test.go]: I don't think you need this helper function

Comment (AkihiroSuda) [reply v2/memoryv2_test.go]: Other parts looks good

File: v2/memoryv2_test.go [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+	assert.Equal(t, "314572800", strings.TrimSpace(string(swapMax)))
+
+	memMax, err := ioutil.ReadFile(filepath.Join(c.path, "memory.max"))
+	if err != nil {
+		t.Fatal("failed to read memory.max file: ", err)
+	}
+	assert.Equal(t, "629145600", strings.TrimSpace(string(memMax)))
+}
+
+func clear(path string) {
```
Comment (fuweid) [inline v2/memoryv2_test.go]: nit: it looks like we can call `os.RemoveAll(path` directly, instead of using helper function.

**Comment by Zyqsempai** [comment]
@fuweid Fixed.

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**